ha/core/event_manager/event_manager.py

```diff
@@ -333,19 +334,37 @@ def get_events(self, component : str) -> list:
         """
         pass
 
-    def publish(self, component:str, event: RecoveryActionEvent) -> None:
+    def publish(self, event: RecoveryActionEvent) -> None:
         """
         Publish event.
         Args:
-            component (str): Component name.
             event (RecoveryActionEvent): Action event.
         """
-        pass
+        try:
+            component_list = []
+            # Run through list of components subscribed for this event and send event to each of them
+            component_list_key = f'{const.EVENT_KEY}/{event.resource_type}/{event.event_type}'
```

Comment:
can we have _prepare_key method to get this keys

Reply:
If you mean the _prepare_key() method in system health module, then I guess lets not it now.
We will have to make it generic before use.